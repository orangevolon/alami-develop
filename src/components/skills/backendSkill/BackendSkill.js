import { Skill, Badge } from "@components";
import nodeBadge from "@assets/badges/node.svg";
import expressBadge from "@assets/badges/express.svg";
import csharpBadge from "@assets/badges/csharp.svg";
import firebaseBadge from "@assets/badges/firebase.svg";
import firestoreBadge from "@assets/badges/firestore.svg";

import "./BackendSkill.scss";

const BackendSkill = () => {
  const nodeBadgeElement = Badge({ src: nodeBadge });
  nodeBadgeElement.className = "node-badge__img";

  const expressBadgeElement = Badge({ src: expressBadge });
  expressBadgeElement.classList.add("express-badge__img");

  const csharpBadgeElement = Badge({ src: csharpBadge });
  csharpBadgeElement.classList.add("csharp-badge__img");

  const firebaseBadgeElement = Badge({ src: firebaseBadge });
  firebaseBadgeElement.classList.add("firebase-badge__img");

  const firestoreBadgeElement = Badge({ src: firestoreBadge });
  firestoreBadgeElement.classList.add("firestore-badge__img");

  const skillElement = Skill({
    badges: [
      nodeBadgeElement,
      expressBadgeElement,
      csharpBadgeElement,
      firebaseBadgeElement,
      firestoreBadgeElement
    ],
    title: "Back End Development",
    description: "Lorem Ipsum"
  });
  skillElement.classList.add("backend-skill__article");

  return skillElement;
};

export default BackendSkill;
