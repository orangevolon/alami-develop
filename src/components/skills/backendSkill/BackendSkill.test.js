import BackendSkill from "./BackendSkill";

describe("Testing components/skill/FrontendSkill", () => {
  it("Should render the component correctly", () => {
    expect(BackendSkill()).toMatchSnapshot();
  });
});
