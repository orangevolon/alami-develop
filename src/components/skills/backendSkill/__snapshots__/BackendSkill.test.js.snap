// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components/skill/FrontendSkill Should render the component correctly 1`] = `
<article
  class="skill__article backend-skill__article"
>
  <h3
    class="title__h3"
  />
  <p
    class="description__p"
  />
  <div
    class="badges__div"
  >
    <img
      class="node-badge__img"
      src="[object Object]"
      style="transform-origin: center center;"
    />
    <img
      class="badge__img express-badge__img"
      src="[object Object]"
      style="transform-origin: center center;"
    />
    <img
      class="badge__img csharp-badge__img"
      src="[object Object]"
      style="transform-origin: center center;"
    />
    <img
      class="badge__img firebase-badge__img"
      src="[object Object]"
      style="transform-origin: center center;"
    />
    <img
      class="badge__img firestore-badge__img"
      src="[object Object]"
      style="transform-origin: center center;"
    />
  </div>
</article>
`;
